mindsdb/integrations/handlers/pinecone_handler/README.md

```diff
@@ -0,0 +1,78 @@
+# Pinecone Handler
+
+This is the implementation of the Pinecone for MindsDB.
+
+## Pinecone
+
+Pinecone is a vector database which is fully-managed, developer-friendly, and easily scalable.
+
+## Implementation
+
+This handler uses `pinecone-client` python library connect to a pinecone environment.
+
+The required arguments to establish a connection are:
+
+* `api_key`: the API key that can be found in your pinecone account
+* `environment`: the environment name corresponding to the `api_key`
+
+## Limitations
+
+- [ ] `CREATE TABLE` support
+    - Creating a table in Pinecone requires 2 additional parameters: dimension and metric
+- [ ] Support for [namespaces](https://docs.pinecone.io/docs/namespaces)
+- [ ] Display score/distance
+- [ ] Support for creating/reading sparse values
+
+## Usage
+
+In order to make use of this handler and connect to an environment, use the following syntax:
+
+```sql
+CREATE DATABASE pinecone_dev
+WITH ENGINE = "pinecone",
+PARAMETERS = {
+   "api_key": "...",
+   "environment": "..."
+};
+```
+
+You can query pinecone indexes (`temp` in the following examples) based on `id` or `search_vector`, but not both:
+
+```sql
+SELECT * from pinecone_dev.temp
+WHERE id = "abc"
+LIMIT 1
+```
+
+```sql
+SELECT * from pinecone_dev.temp
+WHERE search_vector = "[1,2,3,4,5,6,7,8]"
+```
+
+Optionally, you can filter based on metadata too:
+
+```sql
+SELECT * from pinecone_dev.temp
+WHERE id = "abc" AND temp.metadata.hello < 100
+```
+
+You can delete records using `id` or `metadata` like so:
+
+```sql
+SELECT * from pinecone_dev.temp
+WHERE id = "abc"
+```
+
+```sql
+DELETE FROM pinecone_dev.temp
+WHERE temp.metadata.tbd = true
+```
+
+You can insert data into a new collection like so:
+
+```sql
+CREATE TABLE pinecone_dev.temp (
+SELECT * FROM mysql_demo_db.temp LIMIT 10);
+```
+
+To update records, you can use insert statement. When there is a conflicting ID in pinecone index, the record is updated with new values.
```

Comment:
Ideally, show an example of this as well 👍

Reply:
Done!